### PR TITLE
VC GPS Fix

### PR DESCRIPTION
**Pull request by philam2001**
### Summary

- Add new GPS functionality, wanting to use EKF in mode ` SBG_ECOM_SOL_MODE_NAV_POSITION ` to measure vehicle speed
- Fix GPS callback on Quadruna
- Tested and generally works but triggers watchdog sometimes, haven't had time to test it before 2024 Comp so will comment out

### TODO (For the future)
- test ekf while in motion, values at the moment are suspicious
- figure out watchdog issue
### Checklist
*Please change `[ ]` to `[x]` when you are ready.*
- [ ] I have read and followed the code conventions detailed in [README.md](../README.md) (*This will save time for both you and the reviewer!*).
- [ ] If this pull request is longer then **500** lines, I have provided *explicit* justification in the summary above explaining why I *cannot* break this up into multiple pull requests (*Small PR's are faster and less painful for everyone involved!*).

### CODE REVIEW

**Comment by Lucien950** [comment]
@1mackenziekyle after you left we discovered that the pins that you guys made was backwards, cause you need the RX to connect to the TX (because UART pin naming is relative to the broadcasting element). From there we didn't get much further, but we have electrically validated that the GPS is sending the messages to the MCU.

File: can_bus/quadruna/VC/VC_alerts.json [hunk 1]
```diff
@@ -105,6 +105,10 @@
         "RightInverterFault": {
             "id": 556,
             "description": "Right inverter entered fault state."
-        }
+        },
+        "SBGModeFault":  {
+            "id" :561,
+            "description" : "Ensure mode is Nominal mode for EKF data used"
+        }
```
Comment (gtaharaedmonds) [inline can_bus/quadruna/VC/VC_alerts.json:112]: make disabled

**Review by gtaharaedmonds — APPROVED** [review]
Approved once you make that fault disabled too.

File: can_bus/quadruna/VC/VC_alerts.json (hunk 1 above)
Comment (philam2001) [reply can_bus/quadruna/VC/VC_alerts.json]: yup

**Pull request merged**